packages/react/src/components/DateTimePicker/DateTimePickerV2.test.jsx

```diff
@@ -1967,4 +1979,501 @@ describe('DateTimePickerV2', () => {
         .classList.contains(`${iotPrefix}--date-time-picker--trigger-button--disabled`)
     ).toBe(true);
   });
+
+  it('should render in a portal when renderInPortal:true', () => {
+    render(<DateTimePicker {...dateTimePickerProps} renderInPortal />);
+    userEvent.click(screen.getAllByLabelText('Calendar')[0]);
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelectorAll(`#flyout-tooltip`)
+    ).toHaveLength(0);
+  });
+
+  it('should render in a portal when renderInPortal:true (new time spinner)', () => {
+    render(<DateTimePicker {...dateTimePickerProps} useNewTimeSpinner renderInPortal />);
+    userEvent.click(screen.getAllByLabelText('Calendar')[0]);
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelectorAll(`#flyout-tooltip`)
+    ).toHaveLength(0);
+  });
+
+  it('should render as a child when renderInPortal:false', () => {
+    render(<DateTimePicker {...dateTimePickerProps} renderInPortal={false} />);
+    userEvent.click(screen.getAllByLabelText('Calendar')[0]);
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelector('.iot--flyout-menu--tooltip-anchor')
+    ).toHaveAttribute('data-floating-menu-container', 'true');
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelectorAll(`#flyout-tooltip`)
+    ).toHaveLength(1);
+  });
+
+  it('should render as a child when renderInPortal:false (new time spinner)', () => {
+    render(<DateTimePicker {...dateTimePickerProps} useNewTimeSpinner renderInPortal={false} />);
+    userEvent.click(screen.getAllByLabelText('Calendar')[0]);
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelector('.iot--flyout-menu--tooltip-anchor')
+    ).toHaveAttribute('data-floating-menu-container', 'true');
+    expect(
+      screen
+        .getByTestId('date-time-picker-datepicker-flyout-container')
+        .querySelectorAll(`#flyout-tooltip`)
+    ).toHaveLength(1);
+  });
+
+  it('when flyout menu offscreen top left render direction to  RightStart', () => {
+    const originalGetBoundingClientRect = Element.prototype.getBoundingClientRect;
+    const mockGetBoundingClientRect = jest.fn(
+      generateBoundingClientRect({
+        x: -200,
+        y: -200,
+        height: 134,
+        width: 278,
+      })
+    );
```

Comment:
I think the [spec](https://jestjs.io/docs/25.x/jest-object#jestfnimplementation) suggest the following for mock implementation. 
```suggestion
    const mockGetBoundingClientRect = jest.fn(generateBoundingClientRect);
```